tests/trafficmanagement/scripts/request_timeouts_delay.txt

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+source ${REPO_ROOT}/content/en/docs/tasks/traffic-management/request-timeouts/snips.sh
+
+# Export HOST, PORT and URL
+export INGRESS_HOST={{ .ingressHostCom }}
+export INGRESS_PORT={{ .ingressPortCom }}
+export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+
+# Step 1: add a half second request timeout for calls to the reviews service
+snip_request_timeouts_3
+
+istioctl experimental wait --for=distribution VirtualService reviews.default
+
+echo "sending request delay to $INGRESS_HOST:$INGRESS_PORT"
+
+# Step 2: Verify product reviews are unavailable
+# $snippet
+curl -sS http://"${GATEWAY_URL}"/productpage | grep -o "<p>Sorry.*</p>"
+sleep 1
+# $verify verifier="contains"
+<p>Sorry, product reviews are currently unavailable for this book.</p>
+# $endsnippet
```

Comment:
```suggestion
out=$(curl -sS http://"${GATEWAY_URL}"/productpage | grep -o "<p>Sorry.*</p>"  2>&1)
_verify_contains "$out" "<p>Sorry, product reviews are currently unavailable for this book.</p>" "verify_reviews_unavailable"
```

Reply:
If you want to enhance the test to also confirm that it takes about 1 second, you could run the time command and capture the output and then do an extra check that the time is (roughly?) 1 second. There are various ways to get the time, not sure what's best. Here's one idea?
```
{ time out=$(curl -sS http://"${GATEWAY_URL}"/productpage | grep -o "<p>Sorry.*</p>" 2>&1) ; } 2> time.txt
```
This will time the command and put the time in the file `time.txt`